docs/recipes/nunjucks.md

```diff
@@ -101,91 +101,69 @@ gulp.task('serve', () => {
-     .pipe($.if('*.js', $.uglify()))
-     .pipe($.if('*.css', $.cssnano({safe: true, autoprefixer: false})))
-     .pipe($.if('*.html', $.htmlmin({collapseWhitespace: true})))
-     .pipe(gulp.dest('dist'));
- });
+function html() {
+- return src('app/*.html')
++ return src(['app/*.html', '.tmp/*.html'])
+   .pipe($.useref({searchPath: ['.tmp', 'app', '.']}))
+   .pipe($.if(/\.js$/, $.uglify({compress: {drop_console: true}})))
+   .pipe($.if(/\.css$/, $.postcss([cssnano({safe: true, autoprefixer: false})])))
+   .pipe($.if(/\.html$/, $.htmlmin({
+     collapseWhitespace: true,
+     minifyCSS: true,
+     minifyJS: {compress: {drop_console: true}},
+     processConditionalComments: true,
+     removeComments: true,
+     removeEmptyAttributes: true,
+     removeScriptTypeAttributes: true,
+     removeStyleLinkTypeAttributes: true
+  })))
+  .pipe(dest('dist'));
+}
 ```
 
 ### 8. Update `extras`
 
 We don't want to copy over `.njk` files in the build process:
 
 ```diff
- gulp.task('extras', () => {
-   return gulp.src([
-     'app/*.*',
--    '!app/*.html'
-+    '!app/*.html',
-+    '!app/*.njk'
-   ], {
-     dot: true
-   }).pipe(gulp.dest('dist'));
- });
-```
-
-### 9. Configure `wiredep` task to wire Bower components on layout templates only
-
-Wiredep does not support `.njk` ([yet](https://github.com/taptapship/wiredep/pull/258)), so also add in the file type definition.
+function extras() {
+  return src([
+    'app/*',
+-   '!app/*.html'
++   '!app/*.html',
++   '!app/*.njk'
+  ], {
+    dot: true
+  }).pipe(dest('dist'));
+};
+```
+### 9. Edit your `watch` task
 
 ```diff
-  gulp.task('wiredep', () => {
-    ...
-
--   gulp.src('app/*.html')
-+   gulp.src('app/layouts/*.njk')
-      .pipe(wiredep({
-        exclude: ['bootstrap-sass'],
--       ignorePath: /^(\.\.\/)*\.\./
-+       ignorePath: /^(\.\.\/)*\.\./,
-+       fileTypes: {
-+         njk: {
-+           block: /(([ \t]*)<!--\s*bower:*(\S*)\s*-->)(\n|\r|.)*?(<!--\s*endbower\s*-->)/gi,
-+           detect: {
-+             js: /<script.*src=['"]([^'"]+)/gi,
-+             css: /<link.*href=['"]([^'"]+)/gi
-+           },
-+           replace: {
-+             js: '<script src="{{filePath}}"></script>',
-+             css: '<link rel="stylesheet" href="{{filePath}}" />'
-+           }
-+         }
-+       }
-+     }))
--     .pipe(gulp.dest('app'));
-+     .pipe(gulp.dest('app/layouts'));
-  });
-```
+  watch([
+-   'app/*.html',
+    'app/images/**/*',
+    '.tmp/fonts/**/*'
+  ]).on('change', server.reload);
 
++ watch('app/**/*.{html,njk}', views);
+  watch('app/styles/**/*.scss', styles);
+  watch('app/scripts/**/*.js', scripts);
+  watch('app/fonts/**/*', fonts);
+}
+```
 
-### 10. Edit your `serve` task
+### 9. Edit your `serve` task
 
-Edit your `serve` task so that editing `.html` and `.njk` files triggers the `views:reload` task:
+Edit your `serve` task so that editing `.html` and `.njk` files triggers the `viewsreload` function:
```

Comment:
This description needs to be updated to something more accurate:

> Compile `.njk` files before running the server

But instead I suggest deleting this section and moving the code below to step 6, which contains old `gulpfile.js` code and needs to be updated anyway. You can update its title so the whole 6th step section goes something like this:

> ### 6. Add `views` as a dependency of both `serve` and `build`
>
> ```diff
>  if (isDev) {
> -  serve = series(clean, parallel(styles, scripts, fonts), startAppServer);
> +  serve = series(clean, parallel(views, styles, scripts, fonts), startAppServer);
>  } ...
> ```
>
> ```diff
>  const build = series(
>    parallel(
>     lint,
> -   series(parallel(styles, scripts), html),
> +   series(parallel(views, styles, scripts), html),
>     images,
>     fonts,
>     extras
>    ),
>    measureSize
>  );
> ```